src/plugins/field_formats/public/lib/converters/date.ts

```diff
@@ -51,7 +51,7 @@ export class DateFormat extends FieldFormat {
         const date = moment(value);
 
         if (date.isValid()) {
-          return date.format(pattern);
+          return date.locale(i18n.getLocale()).format(pattern);
```

Comment:
This seems to work well, and I confirmed it also works as expected when using a custom date format:
<img width="1997" alt="locale" src="https://github.com/elastic/kibana/assets/25592674/d6ddb79c-be99-4500-a21b-0060460cd67f">